app/models/user.rb

```diff
@@ -0,0 +1,36 @@
+require 'bcrypt'
+
+class User
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :name, String, required: true
+  property :email, String, required: true
+  property :username, Text, required: true
+  property :password_digest, Text
+  #property :name, Text
+
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  attr_reader :password
+
+  validates_uniqueness_of :email
+  validates_uniqueness_of :username
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.